modin/data_management/functions/groupby_function.py

```diff
@@ -148,8 +148,9 @@ def caller(
         numeric_only=True,
         **kwargs,
     ):
-        if not (isinstance(by, (type(query_compiler)) or hashable(by))) or isinstance(
-            by, pandas.Grouper
+        if groupby_args.get("level", None) is None and (
+            not (isinstance(by, (type(query_compiler)) or hashable(by)))
```

Comment:
```suggestion
            not (isinstance(by, (type(query_compiler))) or hashable(by))
```
I think that this is a typo - the way it is currently written is calling isintance on query compiler type or hashable:
```python
isinstance(by, query_compiler_type or hashable_by)
```
but I think what is meant is
```python
isinstance(by, query_compiler_type) or hashable_by
```

Reply:
Yes, good catch